sql/catalyst/src/test/scala/org/apache/spark/sql/types/StructTypeSuite.scala

```diff
@@ -103,4 +104,30 @@ class StructTypeSuite extends SparkFunSuite {
     val interval = "`a` INTERVAL"
     assert(fromDDL(interval).toDDL === interval)
   }
+
+  test("find missing (nested) fields") {
```

Comment:
Could you test the behaviours of unsupported cases (array and map), too?

Reply:
OK, sure.